azure/services/subnets/subnets_test.go

```diff
@@ -338,6 +362,121 @@ func TestReconcileSubnets(t *testing.T) {
 				}).Times(1)
 			},
 		},
+		{
+			name:          "doesn't overwrite existing NAT Gateway",
+			expectedError: "",
+			expect: func(s *mock_subnets.MockSubnetScopeMockRecorder, m *mock_subnets.MockClientMockRecorder) {
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.Subnet("my-subnet").AnyTimes().Return(infrav1.SubnetSpec{
+					ID:         "subnet-id",
+					Name:       "my-subnet",
+					Role:       infrav1.SubnetNode,
+					CIDRBlocks: []string{"10.0.0.0/16"},
+					NatGateway: infrav1.NatGateway{
+						ID:   azure.NatGatewayID("123", "my-rg", "existing-natgateway"),
+						Name: "existing-natgateway",
+						NatGatewayIP: infrav1.PublicIPSpec{
+							Name: "existing-natgateway-ip-name",
+						},
+					},
+				})
+				s.SubnetSpecs().AnyTimes().Return([]azure.SubnetSpec{
+					{
+						Name:              "my-subnet",
+						CIDRs:             []string{"10.0.0.0/16"},
+						VNetName:          "my-vnet",
+						RouteTableName:    "my-subnet_route_table",
+						SecurityGroupName: "my-sg",
+						Role:              infrav1.SubnetNode,
+					},
+				})
+				s.Vnet().AnyTimes().Return(&infrav1.VnetSpec{Name: "my-vnet"})
+				s.ClusterName().AnyTimes().Return("fake-cluster")
+				s.SubscriptionID().AnyTimes().Return("123")
+				s.ResourceGroup().AnyTimes().Return("my-rg")
+				m.Get(gomockinternal.AContext(), "", "my-vnet", "my-subnet").
+					Return(network.Subnet{
+						ID:   to.StringPtr("subnet-id"),
+						Name: to.StringPtr("my-subnet"),
+						SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+							AddressPrefix: to.StringPtr("10.0.0.0/16"),
+							RouteTable: &network.RouteTable{
+								ID:   to.StringPtr("rt-id"),
+								Name: to.StringPtr("my-subnet_route_table"),
+							},
+							NetworkSecurityGroup: &network.SecurityGroup{
+								ID:   to.StringPtr("sg-id"),
+								Name: to.StringPtr("my-sg"),
+							},
+							NatGateway: &network.SubResource{
+								ID: to.StringPtr(azure.NatGatewayID("123", "my-rg", "existing-natgateway")),
+							},
+						},
+					}, nil)
+				s.SetSubnet(infrav1.SubnetSpec{
+					ID:         "subnet-id",
+					Name:       "my-subnet",
+					Role:       infrav1.SubnetNode,
+					CIDRBlocks: []string{"10.0.0.0/16"},
+					NatGateway: infrav1.NatGateway{
+						ID:   azure.NatGatewayID("123", "my-rg", "existing-natgateway"),
+						Name: "existing-natgateway",
+						NatGatewayIP: infrav1.PublicIPSpec{
+							Name: "existing-natgateway-ip-name",
+						},
+					},
+				}).Times(1)
+			},
+		},
+		{
+			name:          "spec has empty CIDR and ID data",
```

Comment:
@CecileRobertMichon PTAL

Reply:
not sure I understand what scenario this is testing

Reply:
> We could also test that if CIDRBlocks and ID are empty in the existing spec, it will set them with the Azure subnet values.

Reply:
ah, sorry for the confusion

what I meant there was infrav1.SubnetSpec returned has no ID/CIDR when calling `s.Subnet` and the Azure GET returns the values, which are used in `s.SetSubnet` (which means we're updating the spec to have these values persisted in AzureCluster.Spec)